core/models/src/node/operations.rs

```diff
@@ -225,15 +333,23 @@ impl CloseOp {
     }
 
     pub fn from_bytes(bytes: &[u8]) -> Option<Self> {
```

Comment:
use `Result<.., failure::Error>` for error messages.
There are also a lot of useful macro from failure crate - `bail`, `ensure`